metrics/processors/pod_enricher.go

```diff
@@ -0,0 +1,132 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package processors
+
+import (
+	"net/url"
+	"time"
+
+	"github.com/golang/glog"
+
+	kube_config "k8s.io/heapster/common/kubernetes"
+	"k8s.io/heapster/metrics/core"
+	kube_api "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/client/cache"
+	kube_client "k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/fields"
+	"k8s.io/kubernetes/pkg/labels"
+)
+
+type PodEnricher struct {
+	podLister *cache.StoreToPodLister
+	reflector *cache.Reflector
+}
+
+func (this *PodEnricher) Name() string {
+	return "pod_enricher"
+}
+
+func (this *PodEnricher) Process(batch *core.DataBatch) (*core.DataBatch, error) {
+	pods, err := this.podLister.List(labels.Everything())
+	if err != nil {
+		return nil, err
+	}
+	for _, pod := range pods {
+		addPodInfo(pod, batch)
+	}
+	return batch, nil
+}
+
+func addPodInfo(pod *kube_api.Pod, batch *core.DataBatch) {
+	podMs, found := batch.MetricSets[core.PodKey(pod.Namespace, pod.Name)]
+	if !found {
+		glog.V(3).Infof("Pod not found: %s", core.PodKey(pod.Namespace, pod.Name))
+		return
+	}
+
+	// Add UID
+	podMs.Labels[core.LabelPodId.Key] = string(pod.UID)
+
+	// Add cpu requests
+	cpuRequestMilli := int64(0)
```

Comment:
How about just setting cpu/mem request for containers and then let the PodAggregator to do its work?